zerver/lib/outgoing_webhook.py

```diff
@@ -25,13 +26,18 @@
     get_user_profile_by_id,
 )
 
+OUTGOING_WEBHOOK_TIMEOUT_SECONDS = 10
```

Comment:
Do we want to make this a setting in `default_settings.py`?  I can imagine someone wanting to tune this, and then we can also do `with self.settings(OUTGOING_WEBHOOK_TIMEOUT_SECONDS=0)` in tests.

Reply:
I'm not 100% convinced that there's much utility to making this configurable -- if folks regularly have >10s timeouts, they're going to have queue backups, most probably.  But sure.

`with self.settings(OUTGOING_WEBHOOK_TIMEOUT_SECONDS=0)` doesn't work for tests because `responses` never lets the timeout fire.  There are existing tests that if it raises a timeout, a reasonable thing happens, and I manually tested that this technique does fire a timeout if the response takes >10s to come back.

Reply:
> if folks regularly have >10s timeouts, they're going to have queue backups, most probably

Yes, though we are probably going to end up making this queue worker async at some point, which would change the calculus for this.